prime-router/src/test/kotlin/azure/TokenFunctionTests.kt

```diff
@@ -0,0 +1,389 @@
+
+class MockHttpResponseMessage : HttpResponseMessage.Builder, HttpResponseMessage {
+    var httpStatus: HttpStatusType = HttpStatus.OK
+    var content: Any? = null
+
+    override fun getStatus(): HttpStatusType {
+        return this.httpStatus
+    }
+
+    override fun getHeader(var1: String): String {
+        return "world"
+    }
+
+    override fun getBody(): Any? {
+        return this.content
+    }
+
+    override fun status(status: HttpStatusType): HttpResponseMessage.Builder {
+        this.httpStatus = status
+        return this
+    }
+
+    override fun header(key: String, value: String): HttpResponseMessage.Builder {
+        return this
+    }
+
+    override fun body(body: Any?): HttpResponseMessage.Builder {
+        this.content = body
+        return this
+    }
+
+    override fun build(): HttpResponseMessage {
+        return this
+    }
+}
+
+class MockHttpRequestMessage : HttpRequestMessage<String?> {
+    val httpHeaders = mutableMapOf<String, String>()
+    val parameters = mutableMapOf<String, String>()
+
+    override fun getUri(): URI {
+        return URI.create("http://foo.com/")
+    }
+
+    override fun getHttpMethod(): HttpMethod {
+        return HttpMethod.GET
+    }
+
+    override fun getHeaders(): Map<String, String> {
+        return this.httpHeaders
+    }
+
+    override fun getQueryParameters(): Map<String, String> {
+        return this.parameters
+    }
+
+    override fun getBody(): String? {
+        return null
+    }
+
+    override fun createResponseBuilder(var1: HttpStatus): HttpResponseMessage.Builder {
+        return MockHttpResponseMessage().status(var1)
+    }
+
+    override fun createResponseBuilder(var1: HttpStatusType): HttpResponseMessage.Builder {
+        return MockHttpResponseMessage().status(var1)
+    }
+}
+
+class MockSettings : SettingsProvider {
+    var organizationStore: MutableMap<String, Organization> = mutableMapOf()
+    var receiverStore: MutableMap<String, Receiver> = mutableMapOf()
+    var senderStore: MutableMap<String, Sender> = mutableMapOf()
+
+    override val organizations get() = this.organizationStore.values
+    override val senders get() = this.senderStore.values
+    override val receivers get() = this.receiverStore.values
+
+    override fun findOrganization(name: String): Organization? {
+        return organizationStore[name]
+    }
+
+    override fun findReceiver(fullName: String): Receiver? {
+        return receiverStore[fullName]
+    }
+
+    override fun findSender(fullName: String): Sender? {
+        return senderStore[Sender.canonicalizeFullName(fullName)]
+    }
+
+    override fun findOrganizationAndReceiver(fullName: String): Pair<Organization, Receiver>? {
+        val (organizationName, _) = Receiver.parseFullName(fullName)
+        val organization = organizationStore[organizationName] ?: return null
+        val receiver = receiverStore[fullName] ?: return null
+        return Pair(organization, receiver)
+    }
```

Comment:
[sonarcloud is claiming](https://sonarcloud.io/component_measures?id=CDCgov_prime-data-hub&metric=new_duplicated_lines_density&pullRequest=2047&selected=CDCgov_prime-data-hub%3Aprime-router%2Fsrc%2Ftest%2Fkotlin%2Fazure%2FTokenFunctionTests.kt&view=list) that these lines are duplicates of those in [FileSettings.kt](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/main/kotlin/FileSettings.kt), I'm not sure how to tell it that they're not.. or at least that adjusting the inheritance tree is a much worse code smell then this duplication.

Reply:
Yeah, I'd expect to see duplication in our test code.